hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/contracts/operations/HederaEvmCreateOperation.java

```diff
@@ -44,9 +44,16 @@ protected boolean isEnabled() {
 
     @Override
     protected Address targetContractAddress(MessageFrame frame) {
-        final var updater = (HederaEvmWorldUpdater) frame.getWorldUpdater();
-        final Address address = updater.newContractAddress(frame.getRecipientAddress());
+        final var sourceAddressOrAlias = frame.getRecipientAddress();
+        final var sourceNonce =
+                frame.getWorldUpdater().getAccount(sourceAddressOrAlias).getNonce();
+        final var updater = (HederaEvmStackedWorldUpdater) frame.getWorldUpdater();
+        // Decrement nonce by 1 to normalize the effect of transaction execution
+        final var alias = Address.contractAddress(sourceAddressOrAlias, sourceNonce - 1L);
+
+        final Address address = updater.newAliasedContractAddress(sourceAddressOrAlias, alias);
         frame.warmUpAddress(address);
-        return address;
+        frame.warmUpAddress(alias);
+        return alias;
```

Comment:
(This business of alias vs address is tricky and I'll bet we'll eventually wish we had a single type that encapsulated both and somehow made it easier to _not_ get them confused and use/return the wrong one in various places.)